### PR TITLE
VEText Migration - DB update to support message service id

### PR DESCRIPTION
**Pull request by EvanParish**
[#746](https://github.com/department-of-veterans-affairs/notification-api/issues/746)
Added sms_sender_service_id column to service_sms_senders table.

### CODE REVIEW

File: app/models.py [hunk 1]
```diff
@@ -534,6 +534,7 @@ class ServiceSmsSender(db.Model):
 
     id = db.Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
     sms_sender = db.Column(db.String(12), nullable=False)
+    sms_sender_service_id = db.Column(db.JSON())
```
Comment (kalbfled) [inline app/models.py:537]: JSON is an unusual value for an ID.  I would expect this to be an integer, string, or UUID.  @k-macmillan says:

> The point of the field is to provide any additional information the service provider may want. In the case of twilio it's one string. In the future there may be others. To allow for that flexibility we're going with a json object.

Please document the basis for this decision in the code.  The class's docstring seems like a good place.

**Review by kalbfled — CHANGES_REQUESTED** [review]

File: app/models.py (hunk 1 above)
Comment (EvanParish) [reply app/models.py:543]: Added documentation to explain reason for JSON.

**Review by kalbfled — DISMISSED** [review]

File: app/models.py [hunk 2]
```diff
@@ -530,10 +530,17 @@ def serialize_service():
...
+    Define the service_sms_senders table
+
+    sms_sender_service_id - this field is a placeholder for any service provider we might want to use. Since different
+    services will have different formats and information, we are using the json type.
+    """
     __tablename__ = "service_sms_senders"
 
     id = db.Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
     sms_sender = db.Column(db.String(12), nullable=False)
+    sms_sender_service_id = db.Column(db.JSON())
```
Comment (k-macmillan) [inline app/models.py:543]:
```suggestion
    sms_sender_specifics = db.Column(db.JSON())
```

File: migrations/versions/0350_add_sms_sender_service_id.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+"""
+from alembic import op
+import sqlalchemy as sa
+
+revision = '0350_add_sms_sender_service_id'
+down_revision = '0349_delete_old_opt_out_function'
+
+
+def upgrade():
+    op.add_column('service_sms_senders', sa.Column('sms_sender_service_id', sa.JSON(), nullable=True))
```
Comment (k-macmillan) [inline migrations/versions/0350_add_sms_sender_service_id.py:16]:
```suggestion
    op.add_column('service_sms_senders', sa.Column('sms_sender_specifics', sa.JSON(), nullable=True))
```

File: migrations/versions/0350_add_sms_sender_service_id.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+
+def downgrade():
+    op.drop_column('service_sms_senders', 'sms_sender_service_id')
```
Comment (k-macmillan) [inline migrations/versions/0350_add_sms_sender_service_id.py:20]:
```suggestion
    op.drop_column('service_sms_senders', 'sms_sender_specifics')
```

**Review by k-macmillan — DISMISSED** [review]

**Review by maham360 — DISMISSED** [review]

File: app/models.py (hunk 2 above)
Comment (EvanParish) [reply app/models.py]: updated

**Review by kalbfled — APPROVED** [review]

**Review by maham360 — APPROVED** [review]

**Review by k-macmillan — APPROVED** [review]

**Review by kalbfled — APPROVED** [review]
Please update the PR description to include the purpose of the workflow changes.

**Pull request merged**